src/webxdc.rs

```diff
@@ -367,10 +367,54 @@ impl Context {
         if send_now {
             // send update now
             // (also send updates on MessagesState::Failed, maybe only one member cannot receive)
+            if !instance.param.exists(Param::WebxdcPendingSerial) { // TODO: param changes should be protected by a mutex
+                instance.param.set_int(
+                    Param::WebxdcPendingSerial,
+                    status_update_serial.to_u32() as i32,
+                );
+            }
+            instance.param.set(Param::LastSubject, descr);
+            instance.update_param(&self).await;
+
+            if self.ratelimit.write().await.can_send() {
+                let status_update_msg_id = self.flush_status_updates(instance).await?;
+                Ok(status_update_msg_id)
+            } else {
+                let duration_until_can_send = self.ratelimit.write().await.until_can_send();
+                info!(
+                    self,
+                    "webxdc updates got rate limited, waiting for {}",
+                    duration_to_str(duration_until_can_send)
+                );
+                async_std::future::timeout(duration_until_can_send, async {
+                    // TODO: this is not correct timeout() is executed directly (copy+paste error from scheduler :)
+                    // what we really want is that the following lines are exexcuted after the given duration
+                    if let Err(err) = self.flush_status_updates(instance).await {
```

Comment:
@link2xt can you say what a good rust approach would be to have `flush_status_updates()` being executed after `duration_until_can_send`?

Reply:
You can use `async_std::task::spawn` to create a new task, and `async_std::task::sleep` inside of it before calling `flush_status_updates`.

Better approach IMO is to modify long running SMTP loop so it flushes all updates right before sending messages, instead of spawning new tasks that nobody keeps track of.

Reply:
i finally went for the "modify long running SMTP loop" approach as there are several tiny issues with the task::spawn this seems to be even easier overall.